toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -840,20 +851,66 @@ func FormatSinglePartition(partDevPath string, partition configuration.Partition
 		}, totalAttempts, retryDuration)
 		if err != nil {
 			err = fmt.Errorf("could not format partition with type %v after %v retries", fsType, totalAttempts)
+			return "", err
 		}
 
 		_, stderr, err := shell.Execute("swapon", partDevPath)
 		if err != nil {
 			err = fmt.Errorf("failed to execute swapon:\n%v\n%w", stderr, err)
 			return "", err
 		}
+
+		return fsType, nil
+
 	case "":
 		logger.Log.Debugf("No filesystem type specified. Ignoring for partition: %v", partDevPath)
+		return "", nil
+
 	default:
 		return fsType, fmt.Errorf("unrecognized filesystem format: %v", fsType)
 	}
+}
 
-	return
+func getMkfsOptions(fsType string) ([]string, error) {
+	switch fsType {
+	case "xfs":
+		mkfsXfsVersion, err := getMkfsXfsVersion()
+		if err != nil {
+			return nil, err
+		}
+
+		// Feature  xfsprogs   kernel
+		// nrext64  v5.19      v5.19
+		if mkfsXfsVersion.Compare(versioncompare.New("5.19")) >= 0 {
+			// Disable feature that is not supported on older kernel.
+			options := []string{"-i", "nrext64=0"}
```

Comment:
The PR summary says that we're trying to change the xfs options so they are consistent regardless of the host OS. Are we getting `mkfs.xfs` from the host system? If so, doesn't this version-specific logic mean that we *aren't* going to consistently create the filesystem? And if not, why are we checking versions?

What am I missing?

Reply:
This threw me for a loop looking at the compare code....

I think our comment is just wrong in versioncompare.go (or I'm crazy...).

The compare returns `1` if `mkfsXfsVersion` is >= "5.91".

Soo if I'm understanding this right, xfsprogs <= 5.19 did not have the `nrext64` option at all, so if we see our version if higher than 5.19 (ie >= 0) we tell it to never use that feature so that the produced images can still run with the old tools in the produced image.

Reply:
Forgive me for being slow here, but how does the kernel relate here? Are there really *2* problems, i.e., (1) different versions of `mkfs.xfs` create the filesystem with different options, *and* (2) newer versions of `mkfs.xfs` create the filesystem with options that may not be supported on certain kernel versions? If so, which kernel versions are we talking about? The kernel version of the image being generated, or the version of the kernel running on the host?

Reply:
You are correct. In fact, I had a [version of this change](https://github.com/microsoft/azurelinux/tree/user/chrisgun/micXfsFeatures) that took into account: the OS kernel version, the build host kernel version, the version of the mkfs tools, and the version of grub. I wasn't a fan of the complexity of the change. Hence, I created this simpler version.